hera_sim/simulate.py

```diff
@@ -434,4 +435,80 @@ def add_xtalk(self, model='gen_whitenoise_xtalk', bls=None, **kwargs):
+                              'thermal_noise':add_noise,
+                              'rfi_stations':add_rfi,
+                              'rfi_impulse':add_rfi,
+                              'rfi_scatter':add_rfi,
+                              'rfi_dtv':add_rfi
+                              }
+    
+    def run_sim(self, sim_file=None, **sim_params):
+        """
+        Accept a dictionary or YAML file of simulation parameters and add in
+        all of the desired simulation components to the Simulator object.
+
+        Args:
+            sim_file (str): string providing a hook to a YAML file
+            
+            sim_params (dict): dictionary of simulation parameters.
+                it must take the form {model:params}, where model is
+                a string specifying the simulation component to be added
+                and params is a dictionary providing all the keyword
+                arguments and their desired values.
+
+            If both sim_file and sim_params are provided, then this function
+            will raise an AssertionError.
+        """
+
+        # keep track of which components don't use models
+        uses_no_model = []
+        for key, val in self.SIMULATION_COMPONENTS.items():
+            if 'model' not in inspect.signature(val).parameters:
+                uses_no_model.append(key)
+
+        assert sim_file is not None or len(sim_params) is not 0, \
+                'Either a hook to a simulation file or a dictionary of ' + \
+                'simulation parameters must be provided.'
+
+        assert sim_file is None or len(sim_params) is 0, \
+                'Either a simulation configuration file or a dictionary ' + \
+                'of simulation parameters may be passed, but not both. ' + \
+                'Please choose only one of the two to pass as an argument.'
+
+        # if a hook to a simulation file is provided, then read it in
+        if sim_file is not None:
+            with open(sim_file, 'r') as doc:
+                try:
+                    sim_params = yaml.load(doc.read(), Loader=yaml.FullLoader)
+                except:
+                    print('Check your configuration file. Something broke.')
+                    sys.exit()
+
+        for model, params in sim_params.items():
+            assert model in self.SIMULATION_COMPONENTS.keys(), \
+                    'Models must be supported by hera_sim. ' + \
+                    "'{}' is currently not supported.".format(model)
+
+            assert isinstance(params, dict), \
+                    'Values of sim_params must be dictionaries. ' + \
+                    "The values for '{}' do not comply.".format(model)
+
+
+        # everything should be in working order at this point, so let's simulate
+        for model, params in sim_params.items():
+            add_component = self.SIMULATION_COMPONENTS[model]
+            if model in uses_no_model:
```

Comment:
This isn't exactly correct. As of now, there's no guarantee that multiplicative effects will be applied last. @steven-murray, do you know how I can make sure that multiplicative effects are added last?

Reply:
oh yes, good catch. I guess you could order the entries in `SIMULATION_COMPONENTS`?

Reply:
I suppose that could work. I'd have to make some other changes elsewhere, but I think I know what to do.